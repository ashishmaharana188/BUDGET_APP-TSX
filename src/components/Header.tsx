import { NavLink } from "react-router-dom";

const Header = () => (
    <header>
      <h1>Expensify</h1>
      <NavLink to={"/"} className={activeClassName} end>DashBoard</NavLink>
      <NavLink to={"/create"} className={activeClassName}>Create Expense</NavLink>
      <NavLink to={"/edit"} className={activeClassName}>Edit Expense</NavLink>
      <NavLink to={"/help"} className={activeClassName}>Help</NavLink>
    </header>
  );
  const activeClassName = ({isActive}: {isActive: boolean}) => isActive? "nav-is-active": "";

  export default Header;