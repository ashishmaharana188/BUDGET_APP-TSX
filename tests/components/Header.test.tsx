import React from "react";
import { test, expect } from "vitest";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import Header from "../../src/components/Header";

test("should render Header correctly", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <Header />
    </MemoryRouter>
  );

  // Create a snapshot of the rendered component
  expect(asFragment()).toMatchSnapshot();
});
