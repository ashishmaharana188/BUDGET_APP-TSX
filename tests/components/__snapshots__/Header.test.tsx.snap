// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render Header correctly 1`] = `
<DocumentFragment>
  <header>
    <h1>
      Expensify
    </h1>
    <a
      aria-current="page"
      class="nav-is-active"
      data-discover="true"
      href="/"
    >
      DashBoard
    </a>
    <a
      class=""
      data-discover="true"
      href="/create"
    >
      Create Expense
    </a>
    <a
      class=""
      data-discover="true"
      href="/edit"
    >
      Edit Expense
    </a>
    <a
      class=""
      data-discover="true"
      href="/help"
    >
      Help
    </a>
  </header>
</DocumentFragment>
`;
